internal/pkg/manifest/transform.go

```diff
@@ -291,29 +292,35 @@ func (t subnetListOrArgsTransformer) Transformer(typ reflect.Type) func(dst, src
 	}
 }
 
-type unionTransformer[Basic, Advanced any] struct{}
+type unionTransformer struct{}
 
-func (t unionTransformer[Basic, Advanced]) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
-	if typ != reflect.TypeOf(Union[Basic, Advanced]{}) {
+// Transformer returns custom merge logic for union types.
+func (t unionTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
+	// :sweat_smile: https://github.com/golang/go/issues/54393
+	// reflect currently doesn't have support for getting type parameters
+	// or checking if a type is a non-specific instantiation of a generic type
+	// (i.e., no way to tell if the type Union[string, bool] is a Union)
+	isUnion := strings.HasPrefix(typ.String(), "manifest.Union[")
+	if !isUnion {
 		return nil
 	}
 
 	return func(dst, src reflect.Value) error {
-		dstStruct, srcStruct := dst.Interface().(Union[Basic, Advanced]), src.Interface().(Union[Basic, Advanced])
-
-		if srcStruct.IsBasic() {
-			var zero Advanced
-			dstStruct.isAdvanced, dstStruct.Advanced = false, zero
-			dstStruct.isBasic = true
-		} else if srcStruct.IsAdvanced() {
-			var zero Basic
-			dstStruct.isBasic, dstStruct.Basic = false, zero
-			dstStruct.isAdvanced = true
+		isBasic := src.MethodByName("IsBasic").Call(nil)[0].Bool()
+		isAdvanced := src.MethodByName("IsAdvanced").Call(nil)[0].Bool()
```

Comment:
seems totally safe 😱 
I wonder if it it's worthwhile catching the whole block in a defer+recover?
Something like:
1. Recover
2. Return `nil` if we deem it's better to skip or return an error saying the field "could not be overriden please don't override it?"

I'm leaning more towards the error and suggest them not to override route?